From 96995bbe941940f873d2ec2fd4cb8aae342f0283 Mon Sep 17 00:00:00 2001
From: PratikX Kumar <pratikx.kumar@intel.com>
Date: Tue, 3 Feb 2015 03:23:58 +0530
Subject: [PATCH] dc_xpwr_charger: Enable Max Throttling set property

Enable max throttling property in DC_XPWR_CHARGER (Dollar Cove Charger)
to allow the framework to write/read the max throttling states

Change-Id: I3e84058af34e8aef63de12fb9d67a1a84f438dbd
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1059
Signed-off-by: PratikX Kumar <pratikx.kumar@intel.com>
---
 drivers/power/dc_xpwr_charger.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/power/dc_xpwr_charger.c b/drivers/power/dc_xpwr_charger.c
index 33351d5..5a8d0c7 100644
--- a/drivers/power/dc_xpwr_charger.c
+++ b/drivers/power/dc_xpwr_charger.c
@@ -221,6 +221,7 @@ struct pmic_chrg_info {
 	int iterm;
 	int cable_type;
 	int cntl_state;
+	int cntl_state_max;
 	int max_temp;
 	int min_temp;
 	bool online;
@@ -635,8 +636,14 @@ static int pmic_chrg_usb_set_property(struct power_supply *psy,
 		info->cable_type = val->intval;
 		info->psy_usb.type = get_power_supply_type(info->cable_type);
 		break;
+	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT_MAX:
+		info->cntl_state_max = val->intval;
+		break;
 	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT:
-		info->cntl_state = val->intval;
+		if (val->intval < info->cntl_state_max)
+			info->cntl_state = val->intval;
+		else
+			ret = -EINVAL;
 		break;
 	case POWER_SUPPLY_PROP_MAX_TEMP:
 		info->max_temp = val->intval;
@@ -718,12 +725,12 @@ static int pmic_chrg_usb_get_property(struct power_supply *psy,
 	case POWER_SUPPLY_PROP_ENABLE_CHARGER:
 		val->intval = info->is_charger_enabled;
 		break;
+	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT_MAX:
+		val->intval = info->cntl_state_max;
+		break;
 	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT:
 		val->intval = info->cntl_state;
 		break;
-	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT_MAX:
-		val->intval = info->pdata->num_throttle_states;
-		break;
 	case POWER_SUPPLY_PROP_MAX_TEMP:
 		val->intval = info->max_temp;
 		break;
-- 
1.9.1

